Tests/RegexTests/ParseTests.swift

```diff
@@ -2569,6 +2565,11 @@ extension RegexTests {
     diagnosticTest(#"\Z??"#, .notQuantifiable)
     diagnosticTest(#"\G*?"#, .notQuantifiable)
     diagnosticTest(#"\z+?"#, .notQuantifiable)
+    diagnosticTest(#"^*"#, .notQuantifiable)
+    diagnosticTest(#"$?"#, .notQuantifiable)
+    diagnosticTest(#"(?=a)+"#, .notQuantifiable)
+    diagnosticTest(#"(?i)*"#, .notQuantifiable)
+    diagnosticTest(#"(?i:)?"#, .notQuantifiable)
```

Comment:
PCRE seems to allow quantified change matching option groups (just not the isolated case `(?i)`), do we need to ban them? Would you have to nest the group e.g `(?:(?i:...))?` instead?